nullaway/src/main/java/com/uber/nullaway/autofix/Writer.java

```diff
@@ -0,0 +1,140 @@
+package com.uber.nullaway.autofix;
+
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.CompilationUnitTree;
+import com.sun.source.tree.Tree;
+import com.sun.tools.javac.code.Symbol;
+import com.uber.nullaway.Config;
+import com.uber.nullaway.ErrorMessage;
+import com.uber.nullaway.Nullness;
+import com.uber.nullaway.autofix.out.ErrorInfo;
+import com.uber.nullaway.autofix.out.Fix;
+import com.uber.nullaway.autofix.out.MethodInfo;
+import com.uber.nullaway.autofix.out.SeperatedValueDisplay;
+import com.uber.nullaway.autofix.out.TrackerNode;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.Element;
+
+public class Writer {
+  public static final String ERROR = "/tmp/NullAwayFix/errors.csv";
+  public static final String METHOD_INFO = "/tmp/NullAwayFix/method_info.csv";
+  public static final String CALL_GRAPH = "/tmp/NullAwayFix/call_graph.csv";
+  public static final String SUGGEST_FIX = "/tmp/NullAwayFix/fixes.csv";
+  public static final String FIELD_GRAPH = "/tmp/NullAwayFix/field_graph.csv";
+  public static final String DELIMITER = "$*$";
+
+  public static String getDelimiterRegex() {
+    StringBuilder ans = new StringBuilder("(");
+    for (int i = 0; i < DELIMITER.length(); i++) {
+      ans.append("\\").append(DELIMITER.charAt(i));
+    }
+    ans.append(")");
+    return ans.toString();
+  }
+
+  public static void saveFix(Fix fix) {
+    appendToFile(fix, SUGGEST_FIX);
+  }
+
+  public static void saveFieldGraphNode(Tree tree, VisitorState state) {
+    TrackerNode node = new TrackerNode(ASTHelpers.getSymbol(tree), state.getPath());
+    appendToFile(node, FIELD_GRAPH);
+  }
+
+  public static void saveCallGraphNode(Tree tree, VisitorState state) {
```

Comment:
This class has a large number of static methods doing I/O without any synchronization. Additionally, even _with_ synchronization, the paths are hardcoded and would be shared between concurrent `javac` instances. I am surprised these haven't led to race conditions already, but I guess even when I am testing this I am only running autofix on one target at a time...